system messages to deadLetters

patriknw opened the pull request:
I have reviewed old remoting and added a few things to Artery that I found useful. Well, the deadLetters for system messages are critical.

patriknw committed:
send dropped system messages to deadLetters

* publish remote lifecycle event for quarantined

drewhk commented on akka-remote/src/main/scala/akka/remote/artery/Codecs.scala:
```diff
@@ -11,6 +11,7 @@ import akka.serialization.{ Serialization, SerializationExtension }
 import akka.stream._
 import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
 import akka.util.OptionVal
+import akka.remote.RemoteMetricsExtension
```
Hm, this should be superseded by AFR

johanandren replied:
and it is marked as internal API, so sounds like we could just not add it for artery

patriknw replied:
it's still a useful development tool to see your message sizes easy, the api is the config

drewhk commented:
I am a bit unsure of the lifecycle events part. I believe we should not take it from the old remoting. I am not sure we need them at all, except maybe signalling quarantined as people sometimes need that event.

johanandren commented on akka-remote/src/main/scala/akka/remote/BoundAddressesExtension.scala:
```diff
@@ -26,7 +27,9 @@ class BoundAddressesExtension(val system: ExtendedActorSystem) extends Extension
+    system.provider.asInstanceOf[RemoteActorRefProvider].transport match {
+      case r: Remoting        ⇒ r.boundAddresses
+      case a: ArteryTransport ⇒ Map(ArteryTransport.ProtocolName → a.addresses)
+    }
```
Does it really make sense to port this, from what I can see it is only used in the `NettyTransportSpec`, and we don't have multiple protocol names any more, so the API feels a bit specific to the old remoting?

drewhk replied:
If it can be avoided then yes, we should avoid it.

patriknw replied:
it's public api, and I thought it wouldn't hurt to not break user's code

patriknw commented:
rebased and squashed, will merge after validation

@drewhk I reverted the lifecycle events for start and stopped, kept quarantine
Also reverted RemoteMetricsExtension and BoundAddressesExtension.
